rasa/nlu/extractors/crf_entity_extractor.py

```diff
@@ -155,24 +169,25 @@ def _validate_configuration(self) -> None:
             )
 
     @classmethod
-    def required_packages(cls) -> List[Text]:
-        return ["sklearn_crfsuite", "sklearn"]
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> CRFEntityExtractorGraphComponent:
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, model_storage, resource, execution_context)
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-        # checks whether there is at least one
-        # example with an entity annotation
-        if not training_data.entity_examples:
-            logger.debug(
-                "No training examples with entities present. Skip training"
-                "of 'CRFEntityExtractor'."
-            )
-            return
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
+        return ["sklearn_crfsuite", "sklearn"]
 
+    def process_training_data(
```

Comment:
This method is actually no needed during training time as entities are already annotated in the training data

Reply:
It was previously part of training, I only extracted this section into this new method. I think it's required because training works with `CRFTokens`, not `TrainingData` objects 🤔

Reply:
Sure, but can't we just move the code from `process_training_data` to `train` instead and dump `process_training_data`? Having `process_training_data` makes it look like anything after it would need this information

Reply:
Aaa, in other words, keep it as it used to be 😄 The only reason I had extracted was due to the [issue description](https://github.com/RasaHQ/rasa/issues/9506), but will revert this change.